modules/apps/frontend-js/frontend-js-react-web/package.json

```diff
@@ -1,8 +1,10 @@
 {
 	"dependencies": {
 		"@clayui/icon": "3.1.0",
+		"@clayui/popover": "3.27.0",
 		"@clayui/tooltip": "3.4.5",
 		"@liferay/frontend-js-state-web": "*",
+		"axe-core": "^4.2.0",
```

Comment:
Not that this will go in here, but we might want to specify a fixed version so it's easy for us to know what we're using without having to look at the lock file 😉 

When I try to run this, I get an issue because the `crypto` dependency is not being properly packaged... not sure if you've addressed this somehow in your local branch 🤔

Reply:
Hey @izaera I thought that the shims on `frontend-js-node-shims` were not being ignored by `.gitignore`, trying to figure out how it got there 🤔, any tips?

Reply:
All `node_modules` directories are ignored by the global [`.gitignore` file](https://github.com/liferay/liferay-portal/blob/master/modules/.gitignore#L16) but that doesn't mean that you can't `git add` directories even under `node_modules`. Simply run `git add` to force git to track them.

Reply:
Thanks @izaera!

Reply:
Well, I'm going ahead and adding the `crypto` shim in `frontend-js-node-shims` instead of the module where it is needed, I am following the same patterns that I see in the repository, well I am adding `crypto-browserify` in place of `crypto` but I´m getting a lot of errors on the console due to dependency, this isn´t very clear to me because it´s happening, because I can see the dependencies in `crypto-browserify`, something like:

```
Missing dependency 'frontend-js-node-shims$browserify-sign@4.2.1/browser/algorithms.json' of 'frontend-js-node-shims$browserify-sign@4.2.1/algos'
```

I've been thinking about adding just the `crypto` package https://www.npmjs.com/package/crypto which has no dependencies but is deprecated 😬

Reply:
Do we know why crypto is being used and if there's any alternative to using it? I dislike the idea of shimming crypto a bit (being crypto a sensitive package as it is related to security).

Said that, I don't know what is happenning... My hypothesis is that the other shims are self-contained and don't carry dependencies, thus they work. The `frontend-js-node-shims` project has a very non-standard build because of what it does. Basically, it's not the bundler which builds it, but us (developers), that emulate the work of the bundler by creating handcrafted AMD modules.

Also, we wanted to get rid of `frontend-js-node-shims`, but it seems we still need it :thinking: 

Maybe we can do something with webpack (in your project) to avoid using `frontend-js-node-shims`? :thinking: 

@jbalsas @nhpatt @bryceosterhaus  Any opinion on this? I can ellaborate more on the implications of having a `frontend-js-node-shims` module in portal if necessary...

Reply:
Yeah I have a little of that feeling when I saw many dependencies for `crypto-browserify`, well `axe-core` it will try to load this node.js dependency if it exists but it ignores it if it is running in the browser, the problem that loader will fail in that case.

> The frontend-js-node-shims project has a very non-standard build because of what it does. Basically, it's not the bundler which builds it, but us (developers), that emulate the work of the bundler by creating handcrafted AMD modules.

I had this feeling when I was emulating it but I realized that we still loaded the other modules for having on the `package.json` dependencies, if that is not necessary, maybe I can just remove `crypto-browserify`. I can add this only in my module

Reply:
Well, I understand to have `js-node-shims` is bad... Good to know that they want to remove this dependency, it will be much easier to deal with it. For now I added shim inside the new module I created for A11y (`frontend-js-a11y-web`), I added the `crypto` dependency just to work but instead shim is used, so we are not loading the crypto package. See 0545a23c536f2791921d7ed671e61a62f7f44513

I can't get it to work just by adding shim, so I had to add the `crypto` package.

Reply:
Sound like a reasonable workaround. Much better than shimming `crypto` in `frontend-js-node-shims` :roll_eyes: . 

If `crypto` is only used in node, but not in the browser, there could be more ways to fix this (I'm thinking of providing alternate js files for the browser, or even transforming the offending file on the fly with babel) but they are much more complex than putting a local shim for `crypto` so, let's go with that.